src/ValidationManifold.jl

```diff
@@ -260,87 +527,220 @@ function injectivity_radius(M::ValidationManifold, method::AbstractRetractionMet
     return injectivity_radius(M.manifold, method)
 end
 function injectivity_radius(M::ValidationManifold, p; kwargs...)
-    is_point(M, p; error = M.mode, kwargs...)
-    return injectivity_radius(M.manifold, array_value(p))
+    is_point(
+        M,
+        p;
+        error = M.mode,
+        within = injectivity_radius,
+        context = (:Input,),
+        kwargs...,
+    )
+    return injectivity_radius(M.manifold, _value(p))
 end
 function injectivity_radius(
     M::ValidationManifold,
     p,
     method::AbstractRetractionMethod;
     kwargs...,
 )
-    is_point(M, p; error = M.mode, kwargs...)
-    return injectivity_radius(M.manifold, array_value(p), method)
+    is_point(
+        M,
+        p;
+        error = M.mode,
+        within = injectivity_radius,
+        context = (:Input,),
+        kwargs...,
+    )
+    return injectivity_radius(M.manifold, _value(p), method)
 end
 
 function inner(M::ValidationManifold, p, X, Y; kwargs...)
-    is_point(M, p; error = M.mode, kwargs...)
-    is_vector(M, p, X; error = M.mode, kwargs...)
-    is_vector(M, p, Y; error = M.mode, kwargs...)
-    return inner(M.manifold, array_value(p), array_value(X), array_value(Y))
+    is_point(M, p; error = M.mode, within = inner, context = (:Input,), kwargs...)
+    is_vector(M, p, X; error = M.mode, within = inner, context = (:Input,), kwargs...)
+    is_vector(M, p, Y; error = M.mode, within = inner, context = (:Input,), kwargs...)
+    return inner(M.manifold, _value(p), _value(X), _value(Y))
 end
 
-function is_point(M::ValidationManifold, p; kw...)
-    return is_point(M.manifold, array_value(p); kw...)
+"""
+    is_point(M::ValidationManifold, p; kwargs...)
+
+perform [`is_point`](@ref) on a [`ValidationManifold`](@ref),
+where two additional keywords can be used
+
+* `within=nothing` to specify a function from within which this call was issued
+* `context::NTuple{N,Symbol}=NTuple{0,Symbol}()` to specify one or more contexts, this
+  call was issued in. The context `:Point` is added before checking whether the test
+  should be performed
+
+all other keywords are passed on.
+"""
+function is_point(
+    M::ValidationManifold,
+    p;
+    within::Union{Nothing,<:Function} = nothing,
```

Comment:
We could also  have `error = M.mode` kwarg here? It would simplify using this method. And then pass `error` to `is_point`.

Reply:
nice idea.

Reply:
did the same for `_msg` and `is_vector`.